eng/common/scripts/update-docs-ci.ps1

```diff
@@ -0,0 +1,271 @@
+function UpdateCSVBasedCI($pkgs, $ciRepo, $locationInDocRepo){
+  $csvLoc = (Join-Path -Path $ciRepo -ChildPath $locationInDocRepo)
+  
+  if (-not (Test-Path $csvLoc)) {
+    Write-Error "Unable to locate package csv at location $csvLoc, exiting."
+    exit(1)
+  }
+
+  $allCSVRows = Get-Content $csvLoc
+  $visibleInCI = @{}
+
+  # first pull what's already available
+  for ($i=0; $i -lt $allCSVRows.Length; $i++) {
+    $pkgDef = $allCSVRows[$i]
+
+    # get rid of the modifiers to get just the package id
+    $id = $pkgDef.split(",")[1] -replace "\[.*?\]", ""
+
+    $visibleInCI[$id] = $i
+  }
+
+  foreach ($releasingPkg in $pkgs) {
+    $installModifiers = "tfm=netstandard2.0"
+    if ($releasingPkg.IsPrerelease) {
+      $installModifiers += ";isPrerelease=true"
+    }
+    $lineId = $releasingPkg.PackageId.Replace(".","").ToLower()
+
+    if ($visibleInCI.ContainsKey($releasingPkg.PackageId)) {
+      $packagesIndex = $visibleInCI[$releasingPkg.PackageId]
+      $allCSVRows[$packagesIndex] = "$($lineId),[$installModifiers]$($releasingPkg.PackageId)"
+    }
+    else {
+      $newItem = "$($lineId),[$installModifiers]$($releasingPkg.PackageId)"
+      $allCSVRows += ($newItem)
+    }
+  }
+
+  Set-Content -Path $csvLoc -Value $allCSVRows
+}
+
+# a "package.json configures target packages for all the monikers in a Repository, it also has a slightly different
+# schema than the moniker-specific json config that is seen in python and js
+function UpdatePackageJson($pkgs, $ciRepo, $locationInDocRepo, $monikerId){
+  $pkgJsonLoc = (Join-Path -Path $ciRepo -ChildPath $locationInDocRepo)
+  
+  if (-not (Test-Path $pkgJsonLoc)) {
+    Write-Error "Unable to locate package json at location $pkgJsonLoc, exiting."
+    exit(1)
+  }
+
+  $allJsonData = Get-Content $pkgJsonLoc | ConvertFrom-Json
+
+  $visibleInCI = @{}
+
+  for ($i=0; $i -lt $allJsonData[$monikerId].packages.Length; $i++) {
+    $pkgDef = $allJsonData[$monikerId].packages[$i]
+    $visibleInCI[$pkgDef.packageArtifactId] = $i
+  }
+
+  foreach ($releasingPkg in $pkgs) {
+    if ($visibleInCI.ContainsKey($releasingPkg.PackageId)) {
+      $packagesIndex = $visibleInCI[$releasingPkg.PackageId]
+      $existingPackageDef = $allJsonData[$monikerId].packages[$packagesIndex]
+      $existingPackageDef.packageVersion = $releasingPkg.PackageVersion
+    }
+    else {
+      $newItem = New-Object PSObject -Property @{ 
+        packageDownloadUrl = "https://repo1.maven.org/maven2"
+        packageGroupId = $releasingPkg.GroupId
+        packageArtifactId = $releasingPkg.PackageId
+        packageVersion = $releasingPkg.PackageVersion
+        inputPath = @()
+        excludePath = @()
+      }
+
+      $allJsonData[$monikerId].packages += $newItem
+    }
+  }
+
+  Set-Content -Path $pkgJsonLoc -Value ($allJsonData | ConvertTo-Json -Depth 10 | % {$_ -replace "(?m)  (?<=^(?:  )*)", "    " })
```

Comment:
Another line to split. Perhaps a good case for a helper function since this was the 3 instances.